### DISCUSSION TITLE
How would you go about building a application with Orchard Core Framework(no cms)

### DISCUSSION THREAD

**Discussion by MarcBruins** [OrchardCMS/OrchardCore#7594] [General]
The modular nature of orchard is great, and for a new project I would like to take the same approach.

My idea is to use a modular framework(orchard), where developers can create their own modules that can integrate with a single web app. This could for example be as TheAdmin user UI, with different modules enabled in the menu and throughout the system based on the modules configured.

This project is a basically a backend where user can login to manage other services. There is no need for a fancy frontend and it should be a full multi-tenant saas solution.

I'm thinking about setting up a empty ASP.NET core project and then

- Enabling multi tenancy with(https://www.nuget.org/packages/OrchardCore.Tenants/1.0.0-rc2-13450)
- Enabling recipes (https://www.nuget.org/packages/OrchardCore.Recipes/1.0.0-rc2-13450). But how would this integrate with a non CMS solution?
- Using the TheAdmin style user UI or something similar
- Enables modules to be listed in the TheAdmin ui and throughout the application where necessary
- Write custom component in ASP.NET core MVC style(the usage and styling can be the same as orchard)

Would this be viable for orchardcore framework? What are the pro/cons and what steps do i need to take?

**Comment by MarcBruins** [comment]
So this looks like a great place to start: https://github.com/OrchardCMS/OrchardCore.Samples

**Reply by davidgoh** [reply]
Have done quite some deep dive into the code to get an application working without the CMS part, still figuring out how to do so until now.

The recent attempt was trying to reuse some of the following core modules:
- OrchardCore.Admin
- OrchardCore.Tenants

But then there is AdminMenuFilter which requires other stuff like OrchardCore.DisplayManagement, OrchardCore.DataStorage, and so on...which is very much toward the CMS side of things.

**Reply by davidgoh** [reply]
> So this looks like a great place to start: https://github.com/OrchardCMS/OrchardCore.Samples

Will be great if you could share your journey too, perhaps we could both come out from the rabbit holes sooner :)

**Reply by ns8482e** [reply]
> But then there is AdminMenuFilter which requires other stuff like OrchardCore.DisplayManagement, OrchardCore.DataStorage, and so on...which is very much toward the CMS side of things.

Clarifying here ...  `OrchardCore.DisplayManagement` is not CMS specific - it's part of OC Framework for building compossible UI - You can use C# POCO objects

**Comment by AnthonySylvia** [comment]
This is a little old, but helped me and might also help others to understand how to setup the Orchard Core Framework by itself without CMS. [https://www.youtube.com/watch?v=yrQaKv2mxFU](https://www.youtube.com/watch?v=yrQaKv2mxFU)

This video references .NET Core 3.0, however the nuget packages mention in it below support 3.1, 5.0 and 6.0 so you might have to adjust the code this example to fit the requirements of [the new hosting model](https://docs.microsoft.com/en-us/aspnet/core/migration/50-to-60-samples?view=aspnetcore-6.0) depending on your version of .NET but that is not exactly an OrachardCore issue to understand.

For me, the main concept to grasp was that you need a host web app and module web apps.

Install and reference OrchardCore.Module.Targets in the modules and OrchardCore.Application.Targets in the host project respectively.
A common oversight I see and have made myself too is that you also have to reference the modules apps from the host app, even if you don't directly add any code, the reference alone does the trick of connecting them together (see 8:15 into the video).
